components/board/pi/impl/board_test.go

```diff
@@ -28,9 +27,14 @@ func TestPiPigpio(t *testing.T) {
 			{Name: "servo-i", Pin: "22", Type: "servo"},
 		},
 	}
-
-	pp, err := NewPigpio(ctx, board.Named("foo"), &cfg, logger)
-	if errors.Is(err, errors.New("not running on a pi")) {
+	resourceConfig := resource.Config{ConvertedAttributes: &cfg}
+
+	pp, err := NewPigpio(ctx, board.Named("foo"), resourceConfig, logger)
+	// Go's error comparison is shallow, meaning it only checks if two variables point to the same
+	// location in memory. We want to check if the error is due to not running on a raspberry pi,
+	// regardless of whether it's a new copy of the error. Consequently, we can't use `errors.Is`
+	// to compare the error directly to one that we think it should resemble.
```

Comment:
Not sure this comment is useful. It explains why this changed, but perhaps it answers a question that no one will be asking 3 months in the future

Reply:
Probably not, but it is good context for today.